ni_measurementlink_service/session_management/_client.py

```diff
@@ -0,0 +1,291 @@
+"""Session management client class."""
+from __future__ import annotations
+
+import logging
+import threading
+import warnings
+from typing import Iterable, Optional, Sequence, Union
+
+import grpc
+
+from ni_measurementlink_service._channelpool import GrpcChannelPool
+from ni_measurementlink_service._internal.discovery_client import DiscoveryClient
+from ni_measurementlink_service._internal.stubs.ni.measurementlink.sessionmanagement.v1 import (
+    session_management_service_pb2,
+    session_management_service_pb2_grpc,
+)
+from ni_measurementlink_service.session_management._constants import (
+    GRPC_SERVICE_CLASS,
+    GRPC_SERVICE_INTERFACE_NAME,
+)
+from ni_measurementlink_service.session_management._reservation import (
+    MultiSessionReservation,
+    SingleSessionReservation,
+)
+from ni_measurementlink_service.session_management._types import (
+    PinMapContext,
+    SessionInformation,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class SessionManagementClient(object):
```

Comment:
This class is going to be instantiated directly by customers in a code module, but the module name is prefaced with an underscore. If I were a client of this class I would hesitate to use it.

Reply:
If we make the submodules public, then we have to maintain compatibility with the submodules, and we introduce multiple ways to refer to these classes.

These classes are re-exported from `__init__.py` using `__all__`. PEP 8 says that the names declared in `__all__` are public APIs: https://pep8.org/#public-and-internal-interfaces

The API reference documentation shows these classes under `session_management` and hides the private submodules:
![image](https://github.com/ni/measurementlink-python/assets/17914399/01247e69-5431-4579-81db-1304d7c6c7fc)